### PR TITLE
Add edge zone parameter to create network interface cmdlet

### PR DESCRIPTION
**Pull request by wdehrich**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
Test fails
Please record the related test case
```
2021-06-05T00:05:41.8195208Z   [xUnit.net 00:19:46.35]     Commands.Network.Test.ScenarioTests.NetworkInterfaceTests.TestNetworkInterfaceEdgeZone [FAIL]
2021-06-05T00:05:41.8197377Z     X Commands.Network.Test.ScenarioTests.NetworkInterfaceTests.TestNetworkInterfaceEdgeZone [2ms]
2021-06-05T00:05:41.8198596Z EXEC : error Message:  [D:\a\1\s\build.proj]
2021-06-05T00:05:41.8200122Z      System.ArgumentException : Unable to find recorded mock file 'D:\a\1\s\src\Network\Network.Test\bin\Debug\netcoreapp2.1\SessionRecords\Commands.Network.Test.ScenarioTests.NetworkInterfaceTests\TestNetworkInterfaceEdgeZone.json'.
2021-06-05T00:05:41.8201854Z   Parameter name: callerIdentity
2021-06-05T00:05:41.8202761Z     Stack Trace:
2021-06-05T00:05:41.8203922Z        at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
2021-06-05T00:05:41.8205315Z      at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
2021-06-05T00:05:41.8206774Z      at Microsoft.Azure.Commands.TestFx.TestManager.RunTestScript(String[] scripts) in D:\a\1\s\tools\TestFx\TestManager.cs:line 220
2021-06-05T00:05:41.8208548Z      at Commands.Network.Test.ScenarioTests.NetworkInterfaceTests.TestNetworkInterfaceEdgeZone() in D:\a\1\s\src\Network\Network.Test\ScenarioTests\NetworkInterfaceTests.cs:line 139
```

**Comment by wdehrich** [comment]
Hello @msJinLei 

My change has a dependency on a change that is not yet in master, so I cannot create a pull request against that branch.

In the meantime, can you please review my other pull request? (This one has a test session included in it)
[https://github.com/Azure/azure-powershell/pull/15196](https://github.com/Azure/azure-powershell/pull/15196)

**Comment by msJinLei** [comment]
@wdehrich Network-February is merged to master. I retarget your PR to master. Please resolve the conflicts.

**Comment by msJinLei** [comment]
> Hello @msJinLei
> 
> My change has a dependency on a change that is not yet in master, so I cannot create a pull request against that branch.
> 
> In the meantime, can you please review my other pull request? (This one has a test session included in it)
> #15196

Can we close this PR now?

File: src/Network/Network/NetworkInterface/NewAzureNetworkInterfaceCommand.cs [hunk 1]
```diff
@@ -56,7 +56,8 @@ public class NewAzureNetworkInterfaceCommand : NetworkInterfaceBaseCmdlet
 
         [Parameter(
             Mandatory = false,
-            ValueFromPipelineByPropertyName = true)]
+            ValueFromPipelineByPropertyName = true,
```
Comment (msJinLei) [inline src/Network/Network/NetworkInterface/NewAzureNetworkInterfaceCommand.cs:59]: This change will also cause change to New-AzNetworkInterface.md. Please regenerate the file.

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Network/Network/NetworkInterface/NewAzureNetworkInterfaceCommand.cs (hunk 1 above)
Comment (wdehrich) [reply src/Network/Network/NetworkInterface/NewAzureNetworkInterfaceCommand.cs:59]: Regenerated the help file

**Review by msJinLei — APPROVED** [review]

**Pull request merged**